### DISCUSSION TITLE
Issues building prometheus

### DISCUSSION THREAD

**Discussion by redwiz666** [prometheus-operator/kube-prometheus#846] [General]
When trying to build prometheus to look in all namespaces from the customizing link there seems to be an issue with a field reference. Below is the error I'm getting after calling ./build all-namespaces.jsonnet

`./build.sh all-namespaces.jsonnet
+ set -o pipefail
++ pwd
+ PATH=/home/kubernetes-user/my-kube-prometheus/tmp/bin:/home/kubernetes-user/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin
+ rm -rf manifests
+ mkdir -p manifests/setup
+ jsonnet -J vendor -m manifests all-namespaces.jsonnet
+ xargs '-I{}' sh -c 'cat {} | gojsontoyaml > {}.yaml' -- '{}'
RUNTIME ERROR: Field does not exist: config
        vendor/grafana/grafana.libsonnet:52:20-44       thunk from <object <anonymous>>
        vendor/grafana/grafana.libsonnet:52:9-45        object <anonymous>
        all-namespaces.jsonnet:18:70-80 thunk from <$>
        <std>:1293:24-25        thunk from <function <anonymous>>
        <std>:1293:5-33 function <anonymous>
        all-namespaces.jsonnet:18:53-81 $
        all-namespaces.jsonnet:18:1-83
        all-namespaces.jsonnet:18:1-83
        During evaluation
`

My all-namespace.jsonnet is below
`local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') +
           (import 'kube-prometheus/kube-prometheus-all-namespaces.libsonnet') + {
  _config+:: {
    namespace: 'monitoring',

    prometheus+:: {
      namespaces: [],
    },
  },
};

{ ['00namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{ ['0prometheus-operator-' + name]: kp.prometheusOperator[name] for name in std.objectFields(kp.prometheusOperator) } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
`

**Comment by paulfantom** [comment]
I just tried the following code using kube-prometheus from `master`:
```
// File is named `all-namespaces.jsonnet`

local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') +
(import 'kube-prometheus/kube-prometheus-all-namespaces.libsonnet') + {
  _config+:: {
    namespace: 'monitoring',
    prometheus+:: {
      namespaces: [],
    },
  },
};

{ ['00namespace-' + name]: kp.kubePrometheus[name] for name in std.objectFields(kp.kubePrometheus) } +
{ ['0prometheus-operator-' + name]: kp.prometheusOperator[name] for name in std.objectFields(kp.prometheusOperator) } +
{ ['node-exporter-' + name]: kp.nodeExporter[name] for name in std.objectFields(kp.nodeExporter) } +
{ ['kube-state-metrics-' + name]: kp.kubeStateMetrics[name] for name in std.objectFields(kp.kubeStateMetrics) } +
{ ['alertmanager-' + name]: kp.alertmanager[name] for name in std.objectFields(kp.alertmanager) } +
{ ['prometheus-' + name]: kp.prometheus[name] for name in std.objectFields(kp.prometheus) } +
{ ['grafana-' + name]: kp.grafana[name] for name in std.objectFields(kp.grafana) }
```

and it seems to produce correct output when using it as follows:
```bash
jb install
./build.sh all-namespaces.jsonnet
```

From output you provided it seems a problem is somewhere in `vendor/grafana/grafana.libsonnet` file and not in `all-namespaces.jsonnet`.
